client/components/search/index.jsx

```diff
@@ -203,6 +205,11 @@ module.exports = React.createClass( {
 		if ( event.key === 'Escape' ) {
 			this.closeSearch( event );
 		}
+
+		if ( this.props.blurOnEnter && event.which === 13 ) {
```

Comment:
If you look at `themes-search-card/index.jsx` this is only being set in the renderMobile function. Losing focus on <kbd>enter</kbd> is pretty annoying for anyone who's using a keyboard so that's why I opted for being more explicit in this first pass. 

Thanks for the viewport link, I think it should be OK to gate off of the viewport dimensions. The only concern is that some of the larger phones may not trigger this, but I'll try it out and test a few different images. https://design.google.com/devices/